### ISSUE TITLE
Sitronix Touchdriver broken

### ISSUE THREAD

**Issue by maltevesper** [STMicroelectronics/STM32CubeU5#33]
The code to detect touch/no touch should be something along the lines of `State->TouchDetected = data[2] & 0x80;`.
Unfortunately there is wildly convoluted code in the driver which only works, to detect the first press and never registers a release ("unpress").

**Comment by maltevesper** [comment]
Glimpsing over the rest of the code and noticing that there are bits like the following says enough about the software quality.
I can not be bothered to provide PRs/fixes to a codebase that is riddled with incomplete and worse faulty implementations for basic features (Drawing to the display with an RGB Array did not work (#32 ), tocuhscreen does not recognize when the finger is lifted again).

```
    State->TouchX = (((uint32_t)data[2] & SITRONIX_TOUCH_POS_LSB_MASK) << 4);

    /* Send back first ready X position to caller */
    State->TouchX = ((((uint32_t)data[2] & SITRONIX_TOUCH_POS_LSB_MASK) << 4) | ((uint32_t)data[3]));
```

Furthermore, the touch controller supposedly does support multi touch, and the [implementation is not that hard](https://elixir.bootlin.com/linux/latest/source/drivers/input/touchscreen/st1232.c)

It would be really nice if STM would provide better BSPs for their devices. And apparently, there are more places were STM could raise its quality standards: #31 .

**Comment by TOUNSTM** [comment]
ST Internal Reference: 141764

**Comment by TOUNSTM** [comment]
Hello @maltevesper,

First,  we're sorry for the late response. The issue has already been solved in the latest [Release v1.5.0](https://github.com/STMicroelectronics/STM32CubeU5/commit/5ad32958b27c2d447a9438503efa6c2d0ccd8030).

Thank you very much for your report.

With regards,
